core/src/main/java/com/google/zxing/aztec/decoder/Decoder.java

```diff
@@ -142,12 +185,14 @@ private static String getEncodedData(boolean[] correctedBits) {
             latchTable = shiftTable;
           }
         } else {
-          result.append(str);
+          byte[] b = str.getBytes(StandardCharsets.US_ASCII);
+          decodedBytes.write(b, 0, b.length);
           // Go back to whatever mode we had been in
           shiftTable = latchTable;
         }
       }
     }
+    result.append(new String(decodedBytes.toByteArray(), encoding));
```

Comment:
Does this need to check whether the byte array is empty? if it results in an empty string, fine

Reply:
> Does this need to check whether the byte array is empty?

Yes, it works fine with an empty string.